File: practice/algorithms/warmup/diagonal_difference/DiagonalDifference.scala
object DiagonalDifference extends App {
  val lines = io.Source.stdin.getLines().toList
  val n = lines.head.toInt
  val matrix = lines.tail.toArray.map(_.split(" ").map(_.toInt))

  var leftRight = 0
  var rightLeft = 0
  (0 until n).foreach {
    i =>
      leftRight += (matrix(i))(i)
      rightLeft += (matrix(i)(n - i - 1))
  }
  println((leftRight - rightLeft).abs)
}

File: practice/algorithms/warmup/diagonal_difference/diagonal_difference.rb
n = gets.to_i
leftRight = 0
rightLeft = 0
n.times do |i|
  row = gets.to_s.split(" ").map { |s| s.to_i}
  leftRight += row[i]
  rightLeft += row[n - i - 1]
end

puts (leftRight - rightLeft).abs